### ISSUE TITLE
Dock

### ISSUE THREAD

**Issue by ahousseini** [jason-tratta/ProgressScreen#2]
If there is a policy or a profile the dock appears and overlays the ProgressScreen until the process is killed. Any idea how to prevent this behaviour?

**Comment by jason-tratta** [comment]
I'm not sure I follow, could you elaborate a bit more?  Are you saying that if a policy fires the dock appears through the Progress screen?  If so...are your policies adding items to the dock?

**Comment by ahousseini** [comment]
I created a policy which customize the dock with dockutil. After this policy completed the dock stays over the ProgressScreen app. So you can't see the progress bar. The ProgressScreen should get back in front. Hope it's more clear.

**Comment by jason-tratta** [comment]
Yup..thanks. That makes perfect sense. I'll see if I can add a method in the tries to keep the app up front in a future update.

**Comment by ygini** [comment]
Here is a sample code that you should use to manage your full screen state as a real kiosk mode:

```
        SetSystemUIMode(kUIModeAllHidden,
         kUIOptionDisableAppleMenu
         | kUIOptionDisableProcessSwitch
         | kUIOptionDisableForceQuit
         | kUIOptionDisableSessionTerminate);

        [self enterFullScreenMode:[NSScreen mainScreen] withOptions:[NSDictionary dictionaryWithObject:[NSNumber numberWithBool:YES] forKey:NSFullScreenModeAllScreens]];
```

This is Objective-C code from a custom NSView (self self is the NSView).

Using this code for full screen, if I kill the dock in background, the kiosk mode is still maintained.

**Comment by jason-tratta** [comment]
@ygini  Thanks, I'll check this out.

**Comment by loceee** [comment]
This will be a valuable addition to make ProgressScreen more awesomesauce.
